Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -119,8 +189,14 @@ namespace AZ
 
                     auto projectTitle = AZ::Utils::GetProjectDisplayName();
 
-                    m_nativeWindow = AZStd::make_unique<AzFramework::NativeWindow>(projectTitle.c_str(), AzFramework::WindowGeometry(0, 0, 1920, 1080));
+                    // It is important to call this before using r_width, r_height or r_fullscreen
+                    // because at the moment this system component initializes before Legacy System.cpp gets to parse
+                    // command line arguments into cvars.
+                    UpdateCVarsFromCommandLine();
+
+                    m_nativeWindow = AZStd::make_unique<AzFramework::NativeWindow>(projectTitle.c_str(), AzFramework::WindowGeometry(0, 0, r_width, r_height));
                     AZ_Assert(m_nativeWindow, "Failed to create the game window\n");
+                    m_nativeWindow->SetFullScreenState(r_fullscreen);
```

Comment:
how will affect calling `SetFullScreenState` on mobile platforms? Is this suppose to be called on host platforms only?

Reply:
See my comment below.